indexing files by WikiID, adding crosslanguage verbforms and stopwords

aaiijmrtt opened the pull request:
a slight modification of the already existing extract_verbs.sh script. I used sed to remove the <Wikidump headers>. as per the GSOC warmup task: https://github.com/dbpedia/fact-extractor/issues/1..

marfox commented:
Why did you use a different name for the `extract_verbs.sh` script?
I can't see the diff in this way.
Please comply to the name, thanks.

marfox commented:
Same thing applies to your `get_by_ID.py`.
You interfere if you create new files, since that makes the code review impossible.

aaiijmrtt commented:
my apologies. this will not happen again..

to sum up: in extract_verbs.sh, I removed Wikipedia headers and matched with all types of verbs in all languages (adding a few echos to help debug). in get_soccer_player_articles.py, I added a null check after the regular expression search (and comments)..

marfox commented on extract_verbs.sh:
```diff
@@ -8,115 +8,140 @@ set -e
+# download Wikipedia dump
 URL="http://download.wikimedia.org/"$LANGCODE"wiki/latest/"$LANGCODE"wiki-latest-pages-articles.xml.bz2"
-echo "Downloading dump from: $URL"
+echo "[CHECK] downloading Wikipedia dump to " $(pwd)
```
I don't understand why you prepend `[CHECK]` to the echoed messages

aaiijmrtt replied:
to quickly distinguish output Im expecting from my program from the outputs of programs called from within my program. may be removed, as it doesnt do much once you know its working..

marfox commented on extract_verbs.sh:
```diff
@@ -8,115 +8,140 @@ set -e
 fi
-cat extracted/*/* | csplit --suppress-matched -z -f 'corpus/doc_' - '/</doc>/' {*}
+
+# split files and remove headers
```
This behavior is not requested and even harmful.
We will need these headers when extracting a corpus subset.
This applies for instance to `get_soccer_players_articles.py`.
Please comply to the corresponding issue instructions.

aaiijmrtt replied:
Id later planned to map Wikipedia article IDs to extracted document names instead of the sequential csplit numbers. but youre right. as it stands this will not work. will revert back..

marfox commented on extract_verbs.sh:
```diff
@@ -8,115 +8,140 @@ set -e
-find extracted -type f -exec bash -c "cat '{}' | treetagger/cmd/tree-tagger-"$LANGUAGE" | grep VER >> verbs" \;
+
+# extract verbs and sort (note: check system path to tree-tagger directory)
+find corpus -type f -exec bash -c "cat '{}' | tree-tagger/cmd/tree-tagger-"$LANGUAGE" | grep V[[:upper:]*] >> verbs" \;
```
This grep expression is dangerous, as it may extract noise.
Please use the suggested one (or less generic than the one here)

aaiijmrtt replied:
agreed. after going through the tagsets of several languages, the only commonality I found for verbs (and only verbs unless Im mistaken) was the 'V' at the beginning. would you suggest keeping language specific verb tagsets in the script?

marfox commented:
Yes, everything is set up to be language-specific in this script.

marfox commented:
See my comments in pull request #17

aaiijmrtt commented:
extract_verbs.sh: custom 'csplit' routine to save Wikipedia articles with doc_<ID> as filename. added verbs for matching for several languages according to tagsets on TreeTagger website (no regex, no noise, but brittle to changes in TreeTagger). please use bash to execute..

get_soccer_player_articles.py: simplified as only checking against file name is required..

aaiijmrtt commented:
bag_of_words.py: language parameterized by langcode. stopword lists added for several languages in resources/stopwords/ directory. will add more later..

aaiijmrtt commented:
crowdflower_results_into_training_data.py: added to documentation. decided to add a majority_parameter (because 50% agreement may not imply a majority in all situations)..

shady-cs15 commented:
hmmmmmmmmm...

marfox commented:
You are:
1. implementing changes in the behavior which are not requested;
2. trying to tackle multiple open issues (without referencing them) in a single pull request.
Please take a **single** issue, work on it, and submit a pull request for it.
You should motivate potential behavior changes in the conversation **before** implementing them.
Thanks

aaiijmrtt commented:
my apologies. Im new to open source contributions, and unfamiliar of its etiquettes.
1. indexing files by Wikipedia ID for easy access should not be a negative. would you consider it?
2. which of the following single issues would you have me tackle?
(i) extracting TreeTagger verbforms by language specific tagset.
(ii) adding a stopwords courpus by language.
I tried them all to get 'warmed up'. I might have misunderstood the purpose of the exercise. if Im getting in your way, I shall stop altogether.

marfox commented:
On 3/18/15 4:03 PM, Amitrajit Sarkar wrote:
> my apologies. Im new to open source contributions, and unfamiliar of its
> etiquettes.
Don't worry, it's all right. :-)
You just need to be more specific and precise, otherwise everything gets 
confusing.
> 1. indexing files by Wikipedia ID for easy access should not be a
> negative. would you consider it?
You can submit a pull request for that, but this feature is not 
requested in the open issues, so it will have a lower priority.
> 2. which of the following single issues would you have me tackle?
> (i) extracting TreeTagger verbforms by language specific tagset.
This is not an open issue.
> (ii) adding a stopwords courpus by language.
This is open, but somebody is working on it already. You might help her 
or him in the corresponding pull request:
https://github.com/dbpedia/fact-extractor/pull/21
> I tried them all to get 'warmed up'. I might have misunderstood the
> purpose of the exercise. if Im getting in your way, I shall stop altogether.
It's fine to try them all. But when you want to submit a patch, you 
should open one single pull request.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dbpedia/fact-extractor/pull/19#issuecomment-83012219>.
>
